src/pages/RFOX/components/Bridge/BridgeConfirm.tsx

```diff
@@ -0,0 +1,416 @@
+  })
+
+  const allowanceContract = useMemo(() => {
+    if (!quote || quote.isErr()) return
+
+    const tradeQuote = quote.unwrap()
+
+    return tradeQuote.steps[0].allowanceContract
+  }, [quote])
+
+  const { data: allowanceData, isLoading: isAllowanceDataLoading } = useAllowance({
+    assetId: bridgeQuote.sellAssetId,
+    spender: allowanceContract,
+    from: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+  })
+
+  const isApprovalRequired = useMemo(
+    () => bnOrZero(allowanceData).lt(bridgeQuote.bridgeAmountCryptoBaseUnit),
+    [allowanceData, bridgeQuote.bridgeAmountCryptoBaseUnit],
+  )
+
+  const approvalCallData = useMemo(() => {
+    if (!allowanceContract) return
+
+    return encodeFunctionData({
+      abi: erc20ABI,
+      functionName: 'approve',
+      args: [getAddress(allowanceContract), BigInt(bridgeQuote.bridgeAmountCryptoBaseUnit)],
+    })
+  }, [allowanceContract, bridgeQuote.bridgeAmountCryptoBaseUnit])
+
+  const {
+    mutate: sendApprovalTx,
+    isPending: isApprovalMutationPending,
+    isSuccess: isApprovalMutationSuccess,
+    isIdle: isApprovalMutationIdle,
+  } = useMutation({
+    ...reactQueries.mutations.approve({
+      assetId: bridgeQuote.sellAssetId,
+      spender: allowanceContract!,
+      from: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+      amount: bridgeQuote.bridgeAmountCryptoBaseUnit,
+      wallet,
+      accountNumber,
+    }),
+    onSuccess: (txHash: string) => {
+      setApprovalTxHash(txHash)
+      toast({
+        title: translate('modals.send.transactionSent'),
+        description: (
+          <Text>
+            {feeAsset?.explorerTxLink && (
+              <Link href={`${feeAsset.explorerTxLink}${txHash}`} isExternal>
+                {translate('modals.status.viewExplorer')} <ExternalLinkIcon mx='2px' />
+              </Link>
+            )}
+          </Text>
+        ),
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+        position: 'top-right',
+      })
+    },
+  })
+
+  const isGetApprovalFeesEnabled = useMemo(
+    () =>
+      Boolean(
+        isApprovalMutationIdle &&
+          isApprovalRequired &&
+          feeAsset &&
+          feeAssetMarketData &&
+          wallet &&
+          accountNumber !== undefined,
+      ),
+    [
+      accountNumber,
+      feeAsset,
+      feeAssetMarketData,
+      isApprovalMutationIdle,
+      isApprovalRequired,
+      wallet,
+    ],
+  )
+
+  const { data: approvalFees, isLoading: isGetApprovalFeesLoading } = useQuery({
+    ...reactQueries.common.evmFees({
+      value: '0',
+      accountNumber: accountNumber!, // see isGetApprovalFeesEnabled
+      feeAsset: feeAsset!, // see isGetApprovalFeesEnabled
+      feeAssetMarketData: feeAssetMarketData!, // see isGetApprovalFeesEnabled
+      to: fromAssetId(bridgeQuote.sellAssetId).assetReference,
+      from: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+      data: approvalCallData!,
```

Comment:
this one isnt part of `isGetApprovalFeesEnabled` so should be added or handled without non-null assertion

Reply:
Tackled in p3 452d670aa

Reply:
Now clean and type-safety 🎉  https://github.com/shapeshift/web/pull/7020#discussion_r1631768133